src/Casting/ArrayCast.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace WendellAdriel\ValidatedDTO\Casting;
+
+class ArrayCast implements Castable
+{
+    /**
+     * @param  string  $property
+     * @param  mixed  $value
+     * @return array
+     */
+    public function cast(string $property, mixed $value): array
+    {
+        if (is_string($value)) {
+            $jsonDecoded = json_decode($value, true);
```

Comment:
I think that this should use `JSON_THROW_ON_ERROR` - as otherwise an invalid JSON string would result in `[ null ]`.

Reply:
I'm not sure on this one, because if you see the test suite

```php
$result = $castable->cast($this->testProperty, 'Test');
$this->assertIsArray($result);
$this->assertEquals(['Test'], $result);
```

When a "normal" string is passed it will wrap it into an array, applying this change would break the logic and the tests for this. Because this is not meant to convert only JSON strings to array, it could also to wrap other strings into an array.